From 4450419de1d632d0b5610bcf538d1295f7678085 Mon Sep 17 00:00:00 2001
From: Li Lingfeng <lilingfeng3@huawei.com>
Date: Mon, 6 Feb 2023 17:50:37 +0800
Subject: [PATCH 018/500] [Huawei] Revert "[Huawei] io_uring: add switch to
 control io_uring whether to enable"

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188335

-------------------------------

This reverts commit a4707faf5a38f9de12c111429dbdf1b98969dec3.

We need to apply patch 788d0824269bef (io_uring: import 5.15-stable
io_uring) to move io_uring to separate directory and solve
the problem of CVE-2023-0240.
Revert this patch and add it again after patch 788d0824269bef to
reduce conflicts.

Signed-off-by: Li Lingfeng <lilingfeng3@huawei.com>
---
 fs/io_uring.c | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/fs/io_uring.c b/fs/io_uring.c
index 4bf2ae47aeea..ccf4e73edfd2 100644
--- a/fs/io_uring.c
+++ b/fs/io_uring.c
@@ -987,17 +987,6 @@ static struct kmem_cache *req_cachep;
 
 static const struct file_operations io_uring_fops;
 
-
-static int io_uring_enable;
-
-static int __init io_uring_enable_setup(char *str)
-{
-	io_uring_enable = 1;
-	return 1;
-}
-
-__setup("iouring_enable", io_uring_enable_setup);
-
 struct sock *io_uring_get_socket(struct file *file)
 {
 #if defined(CONFIG_UNIX)
@@ -9157,9 +9146,6 @@ SYSCALL_DEFINE6(io_uring_enter, unsigned int, fd, u32, to_submit,
 	int submitted = 0;
 	struct fd f;
 
-	if (io_uring_enable == 0)
-		return -EOPNOTSUPP;
-
 	io_run_task_work();
 
 	if (flags & ~(IORING_ENTER_GETEVENTS | IORING_ENTER_SQ_WAKEUP |
@@ -9659,9 +9645,6 @@ static long io_uring_setup(u32 entries, struct io_uring_params __user *params)
 SYSCALL_DEFINE2(io_uring_setup, u32, entries,
 		struct io_uring_params __user *, params)
 {
-	if (io_uring_enable == 0)
-		return -EOPNOTSUPP;
-
 	return io_uring_setup(entries, params);
 }
 
@@ -9967,9 +9950,6 @@ SYSCALL_DEFINE4(io_uring_register, unsigned int, fd, unsigned int, opcode,
 	long ret = -EBADF;
 	struct fd f;
 
-	if (io_uring_enable == 0)
-		return -EOPNOTSUPP;
-
 	f = fdget(fd);
 	if (!f.file)
 		return -EBADF;
-- 
2.17.1

